File: src/Chia.Client/PageFlexer.fs
namespace Chia

    module PageFlexer =
        open Fable.React
        open Fable.React.Props

        /// pageFlexer
        ///
        /// Stretched to height of the remaining page
        /// Inside the page flexer elemtents can be arranged in the order of:
        ///
        /// pageHeader
        /// pageContent or scrollpageContent
        /// pageFooter
        ///
        /// pageContent is always needed, the other two are optional
        /// You can use one of the "frames" functions to avoid having to write pageFlexer [] [ pageContent [] ] all the time.

        let pageFlexer props content =
            let props:seq<IHTMLProp> = Seq.concat [[ ClassName "page-flexer" ]; props]
            div props content

        let pageHeader props content =
            let props:seq<IHTMLProp> = Seq.concat [[ ClassName "page-header" ]; props]
            div props content

        let pageContent props content =
            let props:seq<IHTMLProp> = Seq.concat [[ ClassName "page-content" ]; props]
            div props content

        /// set to 60px height but may be changed with a Style prop
        let pageFooter props content =
            let props:seq<IHTMLProp> = Seq.concat [[ ClassName "page-footer" ]; props]
            div props content

        let scrollPageContent props content =
            let props:seq<IHTMLProp> = Seq.concat [[ ClassName "scroll-page-content"; Id "epic-style" ]; props]
            div props content


        /// frames
        let frameContent props content =
            pageFlexer props [pageContent [] content]

        let frameHeaderContent props contentH contentC =
            pageFlexer props
                [ pageHeader [] contentH
                  pageContent [] contentC]

        let frameHeaderContentFooter props contentH contentC contentF =
            pageFlexer props
                [ pageHeader [] contentH
                  pageContent [] contentC
                  pageContent [] contentF ]

        let frameScrollContent props content =
            pageFlexer props [scrollPageContent [] content]

        let frameHeaderScrollContent props contentH contentC =
            pageFlexer props
                [ pageHeader [] contentH
                  scrollPageContent [] contentC]

        let frameHeaderScrollContentFooter props contentH contentC contentF =
            pageFlexer props
                [ pageHeader [] contentH
                  scrollPageContent [] contentC
                  pageContent [] contentF ]

File: tests/Client/output/src/Chia.Client/PageFlexer.js
import { concat } from "../../.fable/fable-library.3.0.0-nagareyama-rc-007/Seq.js";
import { HTMLAttr } from "../../.fable/Fable.React.7.0.1/Fable.React.Props.fs.js";
import { empty, ofArray, singleton } from "../../.fable/fable-library.3.0.0-nagareyama-rc-007/List.js";
import * as react from "react";
import { keyValueList } from "../../.fable/fable-library.3.0.0-nagareyama-rc-007/MapUtil.js";

export function pageFlexer(props, content) {
    const props_1 = concat([singleton(new HTMLAttr(64, "page-flexer")), props]);
    return react.createElement("div", keyValueList(props_1, 1), ...content);
}

export function pageHeader(props, content) {
    const props_1 = concat([singleton(new HTMLAttr(64, "page-header")), props]);
    return react.createElement("div", keyValueList(props_1, 1), ...content);
}

export function pageContent(props, content) {
    const props_1 = concat([singleton(new HTMLAttr(64, "page-content")), props]);
    return react.createElement("div", keyValueList(props_1, 1), ...content);
}

export function pageFooter(props, content) {
    const props_1 = concat([singleton(new HTMLAttr(64, "page-footer")), props]);
    return react.createElement("div", keyValueList(props_1, 1), ...content);
}

export function scrollPageContent(props, content) {
    const props_1 = concat([ofArray([new HTMLAttr(64, "scroll-page-content"), new HTMLAttr(99, "epic-style")]), props]);
    return react.createElement("div", keyValueList(props_1, 1), ...content);
}

export function frameContent(props, content) {
    return pageFlexer(props, [pageContent(empty(), content)]);
}

export function frameHeaderContent(props, contentH, contentC) {
    return pageFlexer(props, [pageHeader(empty(), contentH), pageContent(empty(), contentC)]);
}

export function frameHeaderContentFooter(props, contentH, contentC, contentF) {
    return pageFlexer(props, [pageHeader(empty(), contentH), pageContent(empty(), contentC), pageContent(empty(), contentF)]);
}

export function frameScrollContent(props, content) {
    return pageFlexer(props, [scrollPageContent(empty(), content)]);
}

export function frameHeaderScrollContent(props, contentH, contentC) {
    return pageFlexer(props, [pageHeader(empty(), contentH), scrollPageContent(empty(), contentC)]);
}

export function frameHeaderScrollContentFooter(props, contentH, contentC, contentF) {
    return pageFlexer(props, [pageHeader(empty(), contentH), scrollPageContent(empty(), contentC), pageContent(empty(), contentF)]);
}

